core/common/buffer.hpp

```diff
@@ -14,292 +14,264 @@
-                 public boost::less_than_comparable<gsl::span<const uint8_t>> {
+  template <size_t MaxSize>
+  class BufferT : public SLVector<uint8_t, MaxSize>
+      , public boost::equality_comparable<BufferT<MaxSize>>
+  {
    public:
-    static constexpr bool is_static_collection = false;
-    using iterator = std::vector<uint8_t>::iterator;
-    using const_iterator = std::vector<uint8_t>::const_iterator;
-    using value_type = uint8_t;
-    // with this gsl::span can be built from Buffer
-    using pointer = typename std::vector<uint8_t>::pointer;
-    using const_pointer = typename std::vector<uint8_t>::const_pointer;
-    using size_type = typename std::vector<uint8_t>::size_type;
-
-    /**
-     * @brief allocates buffer of size={@param size}, filled with {@param byte}
-     */
-    Buffer(size_t size, uint8_t byte);
+    using Base = SLVector<uint8_t, MaxSize>;
 
-    ~Buffer() = default;
+    BufferT() = default;
 
     /**
      * @brief lvalue construct buffer from a byte vector
      */
-    explicit Buffer(std::vector<uint8_t> v);
-    explicit Buffer(gsl::span<const uint8_t> s);
-
-    Buffer(const uint8_t *begin, const uint8_t *end);
-
-    Buffer() = default;
-    Buffer(const Buffer &b) = default;
-    Buffer(Buffer &&b) noexcept = default;
-    Buffer(std::initializer_list<uint8_t> b);
-
-    Buffer &reserve(size_t size);
-    Buffer &resize(size_t size);
-
-    Buffer &operator=(const Buffer &other) = default;
-    Buffer &operator=(Buffer &&other) noexcept = default;
-
-    Buffer &operator+=(const Buffer &other) noexcept;
-
-    /**
-     * @brief Accessor of byte elements given {@param index} in bytearray
-     */
-    uint8_t operator[](size_t index) const;
-
-    /**
-     * @brief Accessor of byte elements given {@param index} in bytearray
-     */
-    uint8_t &operator[](size_t index);
-
-    /**
-     * @brief Lexicographical comparison of two buffers
-     */
-    bool operator==(const Buffer &b) const noexcept;
-
-    /**
-     * @brief Lexicographical comparison of buffer and vector of bytes
-     */
-    bool operator==(const std::vector<uint8_t> &b) const noexcept;
-
-    /**
-     * @brief Lexicographical comparison of buffer and vector of bytes
-     */
-    bool operator==(gsl::span<const uint8_t> s) const noexcept;
-
-    /**
-     * @brief Lexicographical comparison of two buffers
-     */
-    bool operator<(const Buffer &b) const noexcept;
-
-    bool operator<(gsl::span<const uint8_t> b) const noexcept;
-
-    /**
-     * @brief Iterator, which points to begin of this buffer.
-     */
-    iterator begin();
-
-    /**
-     * @brief Iterator, which points to the element next to the last in this
-     * buffer.
-     */
-    iterator end();
+    explicit BufferT(const typename Base::Base &other) : Base(other) {}
+    explicit BufferT(typename Base::Base &&other) : Base(std::move(other)) {}
+
+    BufferT(const gsl::span<const typename Base::value_type> &s)
```

Comment:
```suggestion
    explicit BufferT(const gsl::span<const typename Base::value_type> &s)
```

Reply:
It left to keep ability to construct Buffer by span implicitly